risc0/zkvm/src/host/api/client.rs

```diff
@@ -349,11 +349,30 @@ impl Client {
         result
     }
 
-    /// Prove the verification of a recursion receipt using the Poseidon254 hash function for FRI.
+    /// Compress a [Receipt], proving the same computation using a smaller representation.
     ///
-    /// The identity_p254 program is used as the last step in the prover pipeline before running the
-    /// Groth16 prover. In Groth16 over BN254, it is much more efficient to verify a STARK that was
-    /// produced with Poseidon over the BN254 base field compared to using Poseidon over BabyBear.
+    /// Proving will, by default, produce a [CompositeReceipt](crate::CompositeReceipt), which
+    /// may contain an arbitrary number of receipts assembled into continuations and compositions.
```

Comment:
nit: replace 'continuations' with 'segments', and maybe 'compositions' with 'assumptions'.